### PR TITLE
[OCTANE] Add edition upgrade guide

### PR DESCRIPTION
**Pull request by pzuraq**
- [x] General introduction and overview
- [x] Creating an app
- [ ] ~Migrating an existing app~
- [ ] ~File Layout~
- [x] Templates
  - [x] Angle Bracket Syntax
  - [x] Named Arguments 
  - [x] Required `this`
- [x] Native Classes
  - [x] `constructor` instead of `init`
  - [x] Fields vs. Properties
  - [x] Getters and Setters
  - [x] Decorators (includes `@action`)
  - [x] `super`
  - [x] `static`
- [x] Tracked Properties
  - [x] POJOs
  - [x] Arrays
  - [x] Backwards Compatibility
- [x] Glimmer Components
  - [x] Outer HTML
  - [x] `...attributes`
  - [x] Arguments
  - [x] One-way Data Flow
  - [x] Modifiers
  - [x] Template-Only Glimmer Components

### CODE REVIEW

File: guides/release/upgrading/editions.md [hunk 1]
```diff
@@ -0,0 +1,1078 @@
+Editions are a new concept that were recently introduced to Ember. An edition
```
Comment (chriskrycho) [inline guides/release/upgrading/editions.md]: A quick observation – it's probably useful to split this introduction to Editions apart from the **Features** section. They have different potential audiences, especially for people who may come in via media pushes.

File: guides/release/upgrading/editions.md [hunk 2]
```diff
@@ -0,0 +1,1078 @@
...
+  },
+});
+```
+
+```hbs {data-filename=blog-post.hbs}
+<!-- This is the original title, "Hello, world!" -->
+<h1>{{this.title}}</h1>
+
+<!-- This is the uppercased title, "HELLO, WORLD!" -->
+<h1>{{@title}}</h1>
```
Comment (kovalchik) [inline guides/release/upgrading/editions.md]: Should these be reversed?

Comment (pzuraq) [reply guides/release/upgrading/editions.md]: Yup, thanks for pointing that out!

**Comment by ddoria921** [comment]
This documentation reads great! It has a friendly tone and gets straight to the point. Great job working on this draft (and to everyone else involved)! 🎉 👏

File: guides/release/upgrading/editions.md (hunk 2 above)
Comment (kovalchik) [reply guides/release/upgrading/editions.md]: No problem. I definitely agree with @ddoria921, really good work. I think this reads clearly and concisely, despite having to cover a *lot* of ground. Well done!

File: guides/release/upgrading/editions.md [hunk 3]
```diff
@@ -0,0 +1,1831 @@
...
+
+```js
+class ShoppingList {
+  @tracked items = [];
+
+  addItem(item) {
+    this.items.push(item);
+
+    // trigger an update
+    this.items = items;
```
Comment (theroncross) [inline guides/release/upgrading/editions.md]: `this.items = this.items`

File: guides/release/upgrading/editions.md [hunk 4]
```diff
@@ -0,0 +1,1831 @@
...
+a prohibitive number of possible properties, or there could be no way to know
+them in advance. In this case, it's recommended that you _reset_ the value
+wherever it is updated:
+
+```js
+class SimpleCache {
+  @tracked _cache = {};
+
+  set(key, value) {
+    this._cache[cacheKey] = cacheValue;
```
Comment (theroncross) [inline guides/release/upgrading/editions.md]: `this._cache[key] = value;`

Comment (pzuraq) [reply guides/release/upgrading/editions.md]: updated, thanks!

**Comment by pzuraq** [comment]
Ready to merge!

**Pull request merged**